x-pack/test/plugin_api_integration/test_suites/event_log/service_api_integration.ts

```diff
@@ -3,9 +3,234 @@
+      const exists = await es.indices.exists({ index: '.kibana-event-log-8.0.0-000001' });
+      expect(exists).to.be.eql(true);
+      expect(configValue).to.be.eql(
+        `--xpack.eventLog.indexEntries=${result.body.isIndexingEntries}`
+      );
+    });
+
+    it('should be able to check if provider actions is registered', async () => {
+      const initResult = await isProviderActionRegistered('provider3', 'action1');
+
+      if (!initResult.body.isProviderActionRegistered) {
+        await registerProviderActions('provider3', ['action1']);
+      }
+      const result1 = await isProviderActionRegistered('provider3', 'action1');
+      expect(result1.body.isProviderActionRegistered).to.be.eql(true);
+
+      const result = await isProviderActionRegistered('provider3', 'action2');
+      expect(result.body.isProviderActionRegistered).to.be.eql(false);
+    });
+
+    it('should return error message if provider is registered', async () => {
+      const initResult = await isProviderActionRegistered('duplication', 'action1');
+
+      if (!initResult.body.isProviderActionRegistered) {
+        await registerProviderActions('duplication', ['action1', 'action2']);
+      }
+
+      const result = await registerProviderActions('duplication', ['action1', 'action2']);
+      expect(result.badRequest).to.be.eql(true);
+    });
+
+    it('should allow to register provider actions and return all provider actions', async () => {
+      const initResult = await isProviderActionRegistered('provider1', 'action1');
+
+      if (!initResult.body.isProviderActionRegistered) {
+        await registerProviderActions('provider1', ['action1', 'action2']);
+      }
+
+      const providerActions = await getProviderActions('provider1');
+      expect(providerActions.body.actions).to.be.eql(['action1', 'action2']);
+    });
+
+    it('should allow to get event logger event log service', async () => {
+      const initResult = await isProviderActionRegistered('provider2', 'action1');
+
+      if (!initResult.body.isProviderActionRegistered) {
+        await registerProviderActions('provider2', ['action1', 'action2']);
+      }
+      const eventLogger = await getEventLogger('provider2');
+      expect(eventLogger.body.eventLogger.initialProperties).to.be.eql({
+        event: { provider: 'provider2' },
+      });
+    });
+
+    it('should allow write an event to index document if indexing entries is enabled', async () => {
+      const initResult = await isProviderActionRegistered('provider4', 'action1');
+
+      if (!initResult.body.isProviderActionRegistered) {
+        await registerProviderActions('provider4', ['action1', 'action2']);
+      }
+
+      const eventId = '1';
+      const event: IEvent = {
+        event: { action: 'action1', provider: 'provider4' },
+        kibana: { saved_objects: [{ type: 'event_log_test', id: eventId }] },
+      };
+      await startTimingEventLogger(event);
+      setTimeout(() => log.debug('doing some action'), 500);
```

Comment:
Are we able to replace with this a retry.try on the code below to avoid flakiness?